### ISSUE TITLE
Proper detection of unused-variable

### ISSUE THREAD

**Issue by pulkin** [pylint-dev/pylint#8574]
### Current problem

For example, pycharm will signal a very basic issue with this code:
```python
def a():
    unused_variable = 'a'  # unused assignment
    unused_variable = 'b'
    print(unused_variable)
```
(not the case with pylint).

### Desired solution

Would be great to have this pattern included in W0612 "unused-variable".

**Comment by jacobtylerwalls** [comment]
Assuming the typo in line 4 is intentional, this raises `unused-variable` after all. If the typo is not intended, isn't the variable used by the print statement?

**Comment by pulkin** [comment]
Fixed the typo.

"unused variable" is just a common name for it. Alternatively, it could be an "assignment with no effect".

**Comment by jacobtylerwalls** [comment]
But it is used in line 4, where it's provided to `print()`.

**Comment by mbyrnepr2** [comment]
Is [5838](https://github.com/pylint-dev/pylint/issues/5838) the one you mean @nickdrozd ?
